### PR TITLE
Update docs of PolyCollection

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Numpy-docification of `PolyCollection` and some related methods.

### CODE REVIEW

File: lib/matplotlib/collections.py [hunk 1]
```diff
@@ -1051,30 +1051,47 @@ class PolyCollection(_CollectionWithSizes):
...
-        taken cyclically from the array.
-
-        *closed*, when *True*, will explicitly close the polygon.
-
-        %(Collection)s
+        Parameters
+        ----------
+        verts : sequence
+            The sequence polygons described as (*verts0*, *verts1*, ...) where
+            each polygon *verts_i* is either a sequence of vertices (*x*, *y*),
```
Comment (anntzer) [inline lib/matplotlib/collections.py]: each polygon verts_i is an array-like of shape (nv, 2)
?

File: lib/matplotlib/collections.py [hunk 2]
```diff
@@ -1051,30 +1051,47 @@ class PolyCollection(_CollectionWithSizes):
...
+            or it is an equivalent :mod:`numpy` array of shape (*nv*, 2).
+        sizes : array-like, default: None
+            Scaling factors for the polygons *verts_i*. The scaling is applied
```
Comment (anntzer) [inline lib/matplotlib/collections.py]: Each polygon is scaled by the square-root of the corresponding entry in sizes (i.e., sizes specify the scaling of areas).  The scaling is applied before the Artist master transform.  If len(sizes) < len(verts), ...

File: lib/matplotlib/collections.py [hunk 3]
```diff
@@ -1051,30 +1051,47 @@ class PolyCollection(_CollectionWithSizes):
...
 
     def set_verts(self, verts, closed=True):
-        '''This allows one to delay initialization of the vertices.'''
+        """
+        Set the vertices of the polygons.
+
+        Parameters
+        ----------
+        verts : sequence
+            The sequence polygons described as (*verts0*, *verts1*, ...) where
```
Comment (anntzer) [inline lib/matplotlib/collections.py]: as above

File: lib/matplotlib/collections.py [hunk 4]
```diff
@@ -1122,13 +1139,18 @@ class BrokenBarHCollection(PolyCollection):
     @docstring.dedent_interpd
     def __init__(self, xranges, yrange, **kwargs):
         """
-        *xranges*
-            sequence of (*xmin*, *xwidth*)
+        Parameters
+        ----------
+        xranges : sequence of (float, float)
+            The horizontal position and extent for each bar given as sequence
```
Comment (anntzer) [inline lib/matplotlib/collections.py]: The sequence of (left-edge-position, width) of each bar.
(no need to introduce xmin, xwidth as variables)

File: lib/matplotlib/collections.py [hunk 5]
```diff
@@ -1122,13 +1139,18 @@ class BrokenBarHCollection(PolyCollection):
...
+            of tuples (*xmin*, *xwidth*).
 
-        *yrange*
-            *ymin*, *ywidth*
+        yrange : (float, float)
+            The vertical position and extent for all the bars given as
```
Comment (anntzer) [inline lib/matplotlib/collections.py]: The (lower-edge, height) common to all bars.

File: lib/matplotlib/patches.py [hunk 6]
```diff
@@ -232,7 +232,7 @@ def contains_points(self, points, radius=None):
                                                radius)
 
     def update_from(self, other):
-        """Updates this `.Patch` from the properties of *other*."""
+        """Update this `.Patch` from the properties of *other*."""
```
Comment (anntzer) [inline lib/matplotlib/patches.py]: docstring is inherited?

File: lib/matplotlib/collections.py (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/collections.py]: Taken a variation of this.

File: lib/matplotlib/collections.py (hunk 2 above)
Comment (timhoffm) [reply lib/matplotlib/collections.py]: Taken a variation of this.

File: lib/matplotlib/collections.py [hunk 7]
```diff
@@ -1051,30 +1051,44 @@ class PolyCollection(_CollectionWithSizes):
...
+        verts : sequence
+            The sequence of polygons [*verts0*, *verts1*, ...] where each
+            element *verts_i* defines the vertices of polygon *i* as a 2D
+            array-like of of shape (M, 2).
+        sizes : array-like, default: None
+            Squared scaling factors for the polygons. The coordinates of each
+            polygon *verts_i* are multiplied by the square-root of the
+            corresponding entry in *sizes* (i.e., *sizes* specify the scaling
+            of areas). The scaling is applied before the Artist master
+            transform. If ``len(sizes) < len(verts)``, the additional values
```
Comment (anntzer) [inline lib/matplotlib/collections.py]: `if *sizes* is shorter than *verts*`

Comment (timhoffm) [reply lib/matplotlib/collections.py]: Taken. :smile:

**Pull request merged**